### PR TITLE
Support torch-free ONNX inference (pip install --no-deps)

### PR DESCRIPTION
**Pull request by EHxuban11**
Closes the ask in discussion #711.

## What

`pip install --no-deps libreyolo` + `numpy` + `onnxruntime` is now a working inference install. No torch wheel needed.

Before: `import libreyolo` died on `models/base/model.py:29 import torch`.

## Why

Deployment users want ONNX inference without paying for the torch wheel, and want to reuse LibreYOLO's pre/postprocessing instead of rewriting it. Reported in #711.

## How

- New `libreyolo/utils/lazy.py`. `LazyModule` proxy resolves a module on first attribute access. Call sites keep writing `torch.foo(...)` unchanged.
- Torch resolved lazily in the modules on the ONNX import path: `backends/base`, `utils/{results,general,image_loader,serialization}`, `postprocess/{yolo9,yolonas,common}`, `models/{yolo9,yolonas,yolox}/utils`.
- New `libreyolo/preprocess/` package, mirroring `libreyolo/postprocess/`. The numpy preprocess recipes move out of `models/`, because importing anything under `models/` runs `models/__init__.py`, which eagerly builds every `nn.Module` to populate the `can_load` registry. `models/*/utils.py` re-export them, so existing import paths still work.
- Core model/results exports in `libreyolo/__init__.py` resolve via `__getattr__`.
- `_build_result` builds `Results` from ndarrays when torch is absent. `Results` containers already accept `TensorLike`. Torch branch unchanged.

## Registry safety

Laziness is at the top level only. `models/__init__.py` stays eager and atomic. Its import ORDER is load-bearing (`can_load` is first-match-wins, so `LibreYOLO9P2` must register before `LibreYOLO9`). Touching any model name imports that whole module in one go, so the registry is never observed half-populated.

Verified registry order is byte-identical to dev.

Do not make the per-family imports in `models/__init__.py` lazy. A half-populated registry silently routes a checkpoint to the wrong family. Comment added in-code.

## Verification

- Exported a yolo9 ONNX, ran `predict()` with `torch` and `torchvision` blocked by an import hook. Boxes, scores, classes bit-identical to the torch path. 25 detections both ways.
- `_batched_nms_numpy` matches `torchvision.ops.batched_nms` exactly over 25 random fixtures.
- Registry order identical to dev.
- Ruff clean on all changed files.

## Tests

`tests/unit/test_import_without_torch.py`, 10 tests. Runs the torch-free path in a subprocess with torch blocked, which is the only way to observe it from a process that already imported torch.

This guard is the point. Any new module-level `import torch` on the ONNX path silently kills the contract, and CI would never notice because CI installs torch.

## Family coverage

**All 12 G0/G1 detect families** run ONNX inference without torch.

G0: yolo9, rfdetr. G1: yolo9_e2e, yolo9_p2, ec, rtdetr, rtdetrv2, rtdetrv4, dfine, deim, deimv2, yolonas.

Verified per family: exported ONNX for each, ran `predict()` with torch and torchvision blocked, compared against the torch path.

```
family      torch  free   verdict
yolo9       10     10     IDENTICAL
yolo9_e2e   10     10     IDENTICAL
yolo9_p2    10     10     IDENTICAL
yolonas     10     10     IDENTICAL
ec          10     10     IDENTICAL
rtdetr      10     10     IDENTICAL
rtdetrv2    10     10     IDENTICAL
rtdetrv4    10     10     IDENTICAL
rfdetr      10     10     IDENTICAL
dfine       10     10     IDENTICAL
deim        10     10     IDENTICAL
deimv2      10     10     IDENTICAL

families compared: 12 | problems: 0
```

rtdetrv2 and rtdetrv4 come free: they subclass rtdetr and dfine.

## Not in this PR

- Tasks other than detect.
- Families outside G0/G1.
- Docs page for the lightweight install. Separate PR.
- `LibreYOLO9("model.onnx")` still needs torch, since the model class is an `nn.Module`. Torch-free users go through `OnnxBackend` directly. Worth calling out in the docs page.

## Code provenance

Original code written for this PR against LibreYOLO's own first-party code. The `libreyolo/preprocess/` modules are existing LibreYOLO code moved verbatim from `libreyolo/models/*/utils.py` (and `models/ec/postprocess.py`) with no functional change. No third-party code was ported, adapted, or introduced, and no GPL, AGPL, LGPL, non-commercial, proprietary, or unknown-license material was used.

Opened by an agent.

<details><summary><h3>Greptile Summary</h3></summary>

This PR makes package and ONNX-backend imports lazy enough to support NumPy/ONNX Runtime inference without installing torch.
- Lazily exposes top-level model and result APIs while preserving model-registry import order.
- Moves NumPy preprocessing recipes outside the torch-dependent model package and keeps compatibility re-exports.
- Adds NumPy-backed backend result construction and torch-blocked inference tests.
</details>

<details><summary><h3>Confidence Score: 4/5</h3></summary>

The PR is not yet safe to merge because ordinary operations on the NumPy-backed results produced by torch-free inference still attempt to import torch.

The previously reported result-container failure remains: helpers such as `_move`, `_cpu`, `_numpy`, and `_slice_first` evaluate `torch.Tensor` before their ndarray branches, so supported torch-free callers can still receive `ModuleNotFoundError`.

**Files Needing Attention:** libreyolo/utils/results.py
</details>

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| libreyolo/__init__.py | Defers model and result exports while preserving eager, ordered model-registry initialization when a model export is accessed. |
| libreyolo/backends/base.py | Adds NumPy fallbacks for preprocessing, runtime blobs, result construction, class filtering, masks, keypoints, and empty detections. |
| libreyolo/utils/results.py | Replaces the eager torch import, but ndarray-backed result operations still resolve the lazy torch proxy. |
| libreyolo/utils/lazy.py | Introduces a deferred module proxy used only with fixed internal module names. |
| tests/unit/test_import_without_torch.py | Exercises blocked-torch imports and ONNX prediction parity, including class filtering, but not ordinary ndarray-backed result transformations. |

</details>

<details open><summary><h3>Flowchart</h3></summary>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart LR
  Import[import libreyolo] --> Lazy[Lazy top-level exports]
  ONNX[OnnxBackend] --> Pre[NumPy preprocess package]
  Pre --> Runtime[ONNX Runtime]
  Runtime --> Post[NumPy postprocessing]
  Post --> Results[NumPy-backed Results]
  Model[Native model access] --> Torch[Torch-dependent models registry]
```
</details>

<sub>Reviews (4): Last reviewed commit: ["Keep the torch-free family test off the ..."](https://github.com/libreyolo/libreyolo/commit/114b59dbd48987f75d4aab46455821699cc88c58) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=51457293)</sub>

**Context used:**

- Knowledge Base — [Model zoo, architectures, and inference backends](https://app.greptile.com/libreyolo/-/custom-context/knowledge-base/libreyolo/libreyolo/-/docs/model-zoo.md)
- Knowledge Base — [Postprocess pipeline](https://app.greptile.com/libreyolo/-/custom-context/knowledge-base/libreyolo/libreyolo/-/docs/postprocess-pipeline.md)

### CODE REVIEW

File: libreyolo/backends/base.py [hunk 1]
```diff
@@ -3459,10 +3518,10 @@ def _build_result(
...
-        conf_t = torch.tensor(max_scores, dtype=torch.float32)
-        cls_t = torch.tensor(class_ids, dtype=torch.float32)
-        obb_t = torch.tensor(obb, dtype=torch.float32) if obb is not None else None
+        boxes_t = _f32(boxes)
+        conf_t = _f32(max_scores)
+        cls_t = _f32(class_ids)
+        obb_t = _f32(obb) if obb is not None else None
 
         if classes is not None and len(boxes_t) > 0:
             cls_mask = torch.zeros(len(cls_t), dtype=torch.bool)
```
[the remark this answers is not in the document]
Comment (EHxuban11) [reply libreyolo/backends/base.py]: Valid, fixed in 46d97172.

Reproduced: `predict(..., classes=[0])` raised ModuleNotFoundError torch-free. Added `_zeros_bool`/`_bool_array` next to `_f32`/`_zeros_f32`, and the class mask now goes through `_to_blob` before indexing masks/keypoints. Keypoints wrap via `_f32` instead of `torch.as_tensor`.

Torch-free `classes=[0]` now returns 8 detections, identical to the torch path. The end-to-end parity test is parametrised over `classes` so the filter branch is covered.
